src/main/scala/org/broadinstitute/dsde/firecloud/service/UserService.scala

```diff
@@ -1,50 +1,64 @@
 package org.broadinstitute.dsde.firecloud.service
 
+import java.util.UUID
+
 import akka.actor.{Actor, Props}
 import akka.pattern._
 import com.typesafe.scalalogging.LazyLogging
-import org.broadinstitute.dsde.firecloud.Application
-import org.broadinstitute.dsde.firecloud.dataaccess.{RawlsDAO, ThurloeDAO}
-import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol.{impUserImportPermission, impTerraPreference}
+import org.broadinstitute.dsde.firecloud.{Application, FireCloudConfig}
+import org.broadinstitute.dsde.firecloud.dataaccess.{GoogleServicesDAO, RawlsDAO, ThurloeDAO}
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol.{impProfileWrapper, impTerraPreference, impUserImportPermission}
 import org.broadinstitute.dsde.firecloud.model.Trial.CreationStatuses
 import org.broadinstitute.dsde.firecloud.model.{FireCloudKeyValue, ProfileWrapper, RequestCompleteWithErrorReport, TerraPreference, UserImportPermission, UserInfo}
 import org.broadinstitute.dsde.firecloud.service.PerRequest.{PerRequestMessage, RequestComplete}
-import org.broadinstitute.dsde.firecloud.service.UserService.{DeleteTerraPreference, GetTerraPreference, ImportPermission, SetTerraPreference}
+import org.broadinstitute.dsde.firecloud.service.UserService.{DeleteTerraPreference, GetAllUserKeys, GetTerraPreference, ImportPermission, SetTerraPreference}
 import org.broadinstitute.dsde.rawls.model.WorkspaceAccessLevels
+import org.parboiled.common.FileUtils
 import spray.http.StatusCodes
 import spray.httpx.SprayJsonSupport
 import spray.json.DefaultJsonProtocol
 
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success, Try}
 
+
+
 object UserService {
   sealed trait UserServiceMessage
 
   val TerraPreferenceKey = "preferTerra"
   val TerraPreferenceLastUpdatedKey = "preferTerraLastUpdated"
+  val AnonymousGroupKey = "anonymousGroup"
+  val ContactEmailKey = "contactEmail"
+  // following lists are governed by this license (https://aaron.mit-license.org/) and come from https://github.com/aaronbassett/Pass-phrase
```

Comment:
Since we're explicitly calling out the licensing here, it would be nice to know for sure if it's compatible with Orch's BSD license.

Reply:
thanks, it's under an MIT license so we're good. I added a NOTICE.txt file with license information for those files, as per https://broadinstitute.atlassian.net/wiki/spaces/DSDE/pages/3014826/DSDE+Open+Source+Licensing